features/323-login-id/api.md

```diff
@@ -0,0 +1,182 @@
+# API Summary
+
+## SDK API
+Following snippets use TypeScript:
+```diff
+- async signup(loginIDs: { [key: string]: string }, password: string, data?: object): Promise<User>;
+- async signupWithUsername(username: string, password: string, data?: object): Promise<User>;
+- async signupWithEmail(email: string, password: string, data?: object): Promise<User>;
++ async signup(
++   loginIDs: { [key: string]: string }[] | { [key: string]: string },
++   password: string, data?: object,
++   realm?: string
++ ): Promise<User>;
++ async signupWithUsername(username: string, password: string, data?: object, realm?: string): Promise<User>;
++ async signupWithEmail(email: string, password: string, data?: object, realm?: string): Promise<User>;
+
+- async login(loginID: { [key: string]: string }, password: string): Promise<User>;
+- async loginWithUsername(username: string, password: string): Promise<User>;
+- async loginWithEmail(email: string, password: string): Promise<User>;
++ async login(loginID: { [key: string]: string } | string, password: string, realm?: string): Promise<User>;
++ async loginWithUsername(username: string, password: string, realm?: string): Promise<User>;
++ async loginWithEmail(email: string, password: string, realm?: string): Promise<User>;
+
++ async requestVerification(loginIDKey: string, loginID: string): Promise<void>;
```

Comment:
@kiootic Sorry I overlooked it before, shall we have:

`requestEmailVerification(loginID: string)` instead?

Reasons are:
1. In the app config, we already have semantic meaning of loginIDKey, so when it is about email addresses for example, we know where to look for and verify the email address is a valid login ID for the users, we don't need the developers to specify the loginIDKey again (and loginID is unique anyway)
2. We have different types of verification (at least email, SMS and shortcode have been thought of), so it is better to make it explicit on function name for which type of verification we are talking about.